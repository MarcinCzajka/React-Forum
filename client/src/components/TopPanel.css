.ui.tabular.menu .item:hover {
    color: #7f7f7f !important;
}