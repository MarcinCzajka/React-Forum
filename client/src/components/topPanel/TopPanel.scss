#topPanel {
    position: relative;
    height: 70px;

    &>div div {
        margin: 17px 30px 17px 0;
        padding: 0 25px 0 25px;
    }

    a {
        display: flex;
        align-items: center;
        margin: 17px 0 17px 30px;
        padding: 0 15px 0 15px;
    }

    div {
        cursor: pointer;
    }

    p {
        display: flex;
        align-items: center;
    }
}

#topPanelUsername {
    top: calc(25% - 10px);
}

@media (min-width: 580px) {
    #topPanelUsername {
        position: absolute !important;
        left: calc(50% - 3.7em) !important;
    }
}

@media (max-width: 400px) {
    #topPanelUsername {
        display: none !important;
    }
}