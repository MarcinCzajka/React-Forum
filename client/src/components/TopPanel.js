import React from 'react';
import { Link } from 'react-router-dom';
import { Menu } from "semantic-ui-react";
import LoginForm from "./LoginForm";
import SignUpForm from './SignUpForm';
import { UserConsumer } from '../contexts/UserContext';
import './TopPanel.css';

class TopPanel extends React.Component {
    constructor(props) {
        super(props);

        this.state = { loggedIn: false };
    }

    handleLogin = (e) => {
        this.setState({
            loggedIn: !this.state.loggedIn
        })
    }

    render() {
        return (
            <UserConsumer>
                {context => (
                    <Menu tabular inverted>
                        <Link to=''>
                            <Menu.Item name={context.pages[0].name} />
                        </Link>
                        {context.loggedIn ? (
                                <Link to='/me'>
                                    <Menu.Item name={context.pages[1].name} />
                                </Link>
                                
                        ) : ''}
                            <Link to='/new'>
                                <Menu.Item name={context.pages[2].name} />
                            </Link>
                        {context.loggedIn ? (
                            <Menu.Item position='right' id='topPanelUsername' className='noHover'>
                                {context.userName.toUpperCase()}
                            </Menu.Item>
                        ) : ''}
                        <Menu.Menu position='right'>
                            <SignUpForm />
                            <LoginForm />
                        </Menu.Menu>
                    </Menu>
                )}
            </UserConsumer>
        )
    }
}

export default TopPanel;