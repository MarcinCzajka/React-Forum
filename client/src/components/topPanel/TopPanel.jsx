import React from 'react';
import { Link } from 'react-router-dom';
import { Menu } from "semantic-ui-react";
import LoginFormContainer from "./login/LoginFormContainer";
import SignUpFormContainer from './signup/SignUpFormContainer';
import { LocaleConsumer } from '../../contexts/LocaleContext';
import { UserConsumer } from '../../contexts/UserContext';
import './TopPanel.scss';

export default props => {
    return (
        <LocaleConsumer>
            {locale => (
                <UserConsumer>
                    {user => (

                        <Menu tabular inverted id='topPanel' className='noMargin'>

                            <Link to=''>
                                <Menu.Item name={locale.feedPageLink} />
                            </Link>

                            {user.loggedIn ? (
                                <Link to='/me'>
                                    <Menu.Item name={locale.aboutMeLink} />
                                </Link>
                                    
                            ) : ''}

                            <Link to='/new'>
                                <Menu.Item name={locale.newPostLink} />
                            </Link>

                            {user.loggedIn ? (
                                <Menu.Item position='right' id='topPanelUsername' className='noHover'>
                                    {user.userName.toUpperCase()}
                                </Menu.Item>
                            ) : ''}

                            <Menu.Menu position='right'>
                                <SignUpFormContainer />
                                <LoginFormContainer />
                            </Menu.Menu>

                        </Menu>

                    )}
                </UserConsumer>
            )}
        </LocaleConsumer>
    )
}