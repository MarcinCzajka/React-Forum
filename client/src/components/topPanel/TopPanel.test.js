import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { create } from "react-test-renderer";
import { shallow } from 'enzyme';
import TopPanel from './TopPanel';
import { UserProvider } from '../../contexts/UserContext';
import { LocaleProvider } from '../../contexts/LocaleContext';
import locale from '../../locale/en/dictionary.json';

jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children);

const context = (loggedIn) => ({
    loggedIn: loggedIn,
    userName: (loggedIn ? 'example name' : undefined),
    pages: [
        {name: 'Page 1'},
        {name: 'Page 2'},
        {name: 'Page 3'}
    ],
    addShowLoginToContext: function(){},
    addShowSignupToContext: function(){}
});


describe('TopPanel component', () => {
    test('renders without crashing', () => {
        shallow(<LocaleProvider value={locale}>
                <Router>
                    <TopPanel />
                </Router>
            </LocaleProvider>
        );
    });

    test('match snapshot w/o user logged in', () => {
        const component = create(<LocaleProvider value={locale}>
                <UserProvider value={context(false)}>
                    <Router>
                        <TopPanel />
                    </Router>
                </UserProvider>
            </LocaleProvider>
        );
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w user logged in', () => {
        const component = create(<LocaleProvider value={locale}>
                <UserProvider value={context(true)}>
                    <Router>
                        <TopPanel />
                    </Router>
                </UserProvider>
            </LocaleProvider>
        );
        expect(component).toMatchSnapshot();
    });

});