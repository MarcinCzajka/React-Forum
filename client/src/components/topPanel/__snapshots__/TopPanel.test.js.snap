// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPanel component match snapshot /w user logged in 1`] = `
<div
  className="ui inverted tabular menu"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="item"
      onClick={[Function]}
    >
      Page 1
    </div>
  </a>
  <a
    href="/me"
    onClick={[Function]}
  >
    <div
      className="item"
      onClick={[Function]}
    >
      Page 2
    </div>
  </a>
  <a
    href="/new"
    onClick={[Function]}
  >
    <div
      className="item"
      onClick={[Function]}
    >
      Page 3
    </div>
  </a>
  <div
    className="right item noHover"
    id="topPanelUsername"
    onClick={[Function]}
  >
    EXAMPLE NAME
  </div>
  <div
    className="right menu"
  >
    
    <a
      className="item"
      onClick={[Function]}
    >
      Log Out
    </a>
  </div>
</div>
`;

exports[`TopPanel component match snapshot w/o user logged in 1`] = `
<div
  className="ui inverted tabular menu"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="item"
      onClick={[Function]}
    >
      Page 1
    </div>
  </a>
  
  <a
    href="/new"
    onClick={[Function]}
  >
    <div
      className="item"
      onClick={[Function]}
    >
      Page 3
    </div>
  </a>
  
  <div
    className="right menu"
  >
    <a
      className="item"
      onClick={[Function]}
    >
      Sign Up
    </a>
    <div
      className="ui page modals dimmer transition visible active"
    >
      <div
        className="ui tiny modal transition visible active"
        style={Object {}}
      >
        <div
          className="header"
        >
          Sign up
        </div>
        <div
          className="content"
        >
          <form
            className="ui small error form"
            onSubmit={[Function]}
          >
            
            <div
              className="required field"
            >
              <label>
                Username
              </label>
              <input
                autoComplete="off"
                onChange={[Function]}
                placeholder="Name"
                value=""
              />
            </div>
            <div
              className="required field"
            >
              <label>
                Email
              </label>
              <input
                autoComplete="new-password"
                onChange={[Function]}
                placeholder="Email"
                value=""
              />
            </div>
            <div
              className="required field"
            >
              <label>
                Password
              </label>
              <input
                autoCapitalize="off"
                autoComplete="new-password"
                autoCorrect="off"
                onChange={[Function]}
                placeholder="Password"
                spellCheck="false"
                type="password"
                value=""
              />
              <input
                autoCapitalize="off"
                autoComplete="new-password"
                autoCorrect="off"
                onChange={[Function]}
                placeholder="Repeat"
                spellCheck="false"
                type="password"
                value=""
              />
            </div>
            <div
              className="field"
            >
              <div
                data-onloadcallbackname="onloadCallback"
                data-verifycallbackname="verifyCallback"
                id="g-recaptcha"
              />
            </div>
            <button
              className="ui large fluid button"
              onClick={[Function]}
              type="submit"
            >
              Sign up
            </button>
          </form>
        </div>
      </div>
    </div>
    <a
      className="item"
      onClick={[Function]}
    >
      Log In
    </a>
    <div
      className="ui page modals dimmer transition visible active"
    >
      <div
        className="ui tiny modal transition visible active"
        style={Object {}}
      >
        <div
          className="header"
        >
          Log in
        </div>
        <div
          className="content"
        >
          <form
            className="ui small error form"
            onSubmit={[Function]}
          >
            
            <div
              className="required field"
            >
              <label>
                Email
              </label>
              <input
                onChange={[Function]}
                placeholder="Email"
                value=""
              />
            </div>
            <div
              className="required field"
            >
              <label>
                Password
              </label>
              <input
                autoCapitalize="off"
                autoComplete="new-password"
                autoCorrect="off"
                onChange={[Function]}
                placeholder="Password"
                spellCheck="false"
                type="password"
                value=""
              />
            </div>
            <div
              className="ui checkbox"
              onChange={[Function]}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
            >
              <input
                checked={false}
                className="hidden"
                readOnly={true}
                tabIndex={0}
                type="checkbox"
              />
              <label>
                Stay logged in
              </label>
            </div>
            <button
              className="ui large fluid button"
              onClick={[Function]}
              style={
                Object {
                  "marginTop": "15px",
                }
              }
              type="submit"
            >
              Sign in
            </button>
            <p>
              Don't have account yet? 
              <span
                className="asLink"
                onClick={[Function]}
              >
                Sign up
              </span>
              !
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
